### PR TITLE
DCS-1784 fix validation for body scan date radios and input

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/views/pages/bodyscans/recordBodyScan.njk [hunk 1]
```diff
@@ -71,23 +71,23 @@
                     {% endset -%}
 
                 {{ govukRadios({
-                    name: "dateType",
+                    name: "userSelectedDate",
```
Comment (louisenorris) [inline server/views/pages/bodyscans/recordBodyScan.njk:74]: Match to error name on line 90 to fix anchor link

File: server/views/pages/bodyscans/recordBodyScan.njk [hunk 2]
```diff
@@ -71,23 +71,23 @@
...
                             text: "Another date",
-                            checked: data.dateType == 'another-date',
+                            checked: data.userSelectedDate == 'another-date',
                             conditional: {
                                 html: anotherDateHtml
                             }
                         }
                     ],
-                    errorMessage: errors | findError("user-selected-date")
+                    errorMessage: errors | findError("userSelectedDate")
```
Comment (louisenorris) [inline server/views/pages/bodyscans/recordBodyScan.njk:90]: Made camelcase to match to radio name and fix anchor link

File: server/views/pages/bodyscans/recordBodyScan.njk [hunk 3]
```diff
@@ -36,7 +36,7 @@
                 <h2 class="govuk-heading-m">Date of body scan</h2>
                     {% set anotherDateHtml %}
                         {{ govukDateInput({
-                            id: "another-date-for-body-scan",
+                            id: "another-date",
```
Comment (louisenorris) [inline server/views/pages/bodyscans/recordBodyScan.njk:39]: Removed `-for-body-scan` to fix anchor links for input fields as now the prefix of `another-date` will match to error names in this section

**Review by connormaglynn — APPROVED** [review]
LGTM 🚀

**Pull request merged**